File: leetcode/1.two-sum/go/twosum_test.go
package twosum

import (
	"fmt"
	"reflect"
	"testing"
)

type testCase struct {
	nums   []int
	target int
	want   []int
}

func TestTwoSum(t *testing.T) {

	cases := []testCase{
		{
			nums:   []int{2, 7, 11, 15},
			target: 9,
			want:   []int{0, 1},
		},
		{
			nums:   []int{3, 2, 4},
			target: 6,
			want:   []int{1, 2},
		},
		{
			nums:   []int{3, 3},
			target: 6,
			want:   []int{0, 1},
		},
		{
			nums:   []int{2, 4, 6},
			target: 22,
			want:   []int{},
		},
		{
			nums:   []int{},
			target: 0,
			want:   []int{},
		},
		{
			nums:   []int{0, 4, 3, 0},
			target: 0,
			want:   []int{0, 3},
		},
	}

	for _, tc := range cases {
		t.Run(fmt.Sprintf("%v -> %d", tc.nums, tc.target), func(t *testing.T) {
			got := twoSum(tc.nums, tc.target)

			if !reflect.DeepEqual(got, tc.want) {
				t.Fatalf("twoSum, %v -> %d, expected %v, got %v", tc.nums, tc.target, tc.want, got)
			}
		})
	}
}

File: leetcode/1.two-sum/ts/twosum.test.ts
import { expect, test } from "vitest";
import twoSum from "./twosum.js";

interface testCase {
    nums: number[],
    target: number,
    want: number[],
}

test("twosum", () => {
    const tcs: testCase[] = [
        { nums: [2, 7, 11, 15], target: 9, want: [0, 1] },
        { nums: [3, 2, 4], target: 6, want: [1, 2] },
        { nums: [3, 3], target: 6, want: [0, 1] },
        { nums: [], target: 0, want: [] },
        { nums: [0, 4, 3, 0], target: 0, want: [0, 3] },
    ]

    for (const tc of tcs)
        expect(twoSum(tc.nums, tc.target)).toEqual(tc.want)
});
